frontend/src/Components/Shared/TreeManager/TreeManager.spec.tsx

```diff
@@ -1,21 +1,461 @@
-import '@testing-library/jest-dom';
-import { shallow } from 'enzyme';
+import { SettingFilled } from '@ant-design/icons';
+import { cleanup, fireEvent, render, screen, waitFor } from '@testing-library/react';
+import { Form } from 'antd';
 import React from 'react';
 import { TreeManager } from './TreeManager';
+import { TreeNodeType, TreeProps } from './TreeManagerTypes';
 
+// mock the useForm hook from AntD form context
+jest.mock('antd/lib/form/Form', () => ({
+  ...jest.requireActual('antd/lib/form/Form'),
+  useForm: () => [{
+    getFieldError: jest.fn(), 
+    getFieldInstance: jest.fn(),
+    getFieldValue: jest.fn(),
+    getFieldWarning: jest.fn(),
+    getFieldsError: jest.fn(),
+    getFieldsValue: jest.fn(),
+    getInternalHooks: jest.fn(),
+    isFieldTouched: jest.fn(),
+    isFieldValidating: jest.fn(),
+    isFieldsTouched: jest.fn(),
+    isFieldsValidating: jest.fn(),
+    resetFields: jest.fn(),
+    scrollToField: jest.fn(),
+    setFields: jest.fn(),
+    setFieldsValue: jest.fn(),
+    submit: jest.fn(),
+    validateFields: jest.fn()
+  }]
+}));
 
 describe('TreeFormField component', () => {
-  const requiredProps = {};
+  const mockedTreeData = [
+    {
+      title: 'node1',
+      key: 'node1',
+      children: [],
+      parent: null,
+      properties: {
+        title: 'node1',
+        description: 'the node 1'
+      },
+      expanded: true,
+      isLeaf: false,
+    },
+    {
+      title: 'leaf1',
+      key: 'leaf1',
+      children: [],
+      parent: null,
+      properties: {
+        title: 'leaf1',
+        description: 'leaf 1 '
+      },
+      expanded: true,
+      isLeaf: true,
+    },
+  ];
 
-  const getComponent = (props?:any) => shallow((
+  const requiredProps = {
+    treeData: mockedTreeData,
+  };
+  
+  const getComponent = (props?:TreeProps) => render(
     <TreeManager
       {...requiredProps}
       {...props}
     />
-  ));
+  );
+  
+  // clears all mock artifacts to avoid transition from values and making all tests clean from the beginning
+  afterEach(() => {
+    jest.clearAllMocks();
+    cleanup();
+  });
+
+  // NOTE: This component loads the tree data as soon as the tree data is ready.
+  // This fetch will trigger a state event (in this case setTreeData()).
+  // Because of this, we need to simulate this behaviour in an asynchronous fashion, hence the waitFor is used for all
+  // the following tests.
+  // After an event that triggers a state chnage, this is also being made.
+  // Without this, the tests will pass but a couple of warnings about state changes should be
+  // encapsulated in an act() method will appear
+  
```

Comment:
this tests the rendering  and then if the returned wrapper is defined
can we stay with the common (terrestris) practice, of testing if the component is defined in a separate test, then rendering?
https://github.com/terrestris/react-geo/blob/master/src/LayerTree/LayerTree.spec.tsx